### PR TITLE
Feat/get st data

### PR DESCRIPTION
**Pull request by victorwiebe**

### CODE REVIEW

**Comment by victorwiebe** [comment]
Is ready for review, but only once #141 is merged as I used that branch to pass build

File: src/procedures/CreateCheckpoint.ts [hunk 1]
```diff
@@ -9,8 +9,16 @@ import {
...
 
-export class CreateCheckpoint extends Procedure<CreateCheckpointProcedureArgs, Checkpoint> {
+export const createRefreshSecurityTokenFactoryResolver = (
+  factories: Factories,
+  securityTokenId: string
+) => async () => {
+  return factories.securityTokenFactory.refresh(securityTokenId);
+};
+
+export class CreateCheckpoint extends Procedure<CreateCheckpointProcedureArgs, Checkpoint | void> {
```
Comment (monitz87) [inline src/procedures/CreateCheckpoint.ts]: Why did you add `void` to the procedure's return type?

File: src/procedures/IssueTokens.ts [hunk 2]
```diff
@@ -11,8 +11,16 @@ import {
...
 
-export class IssueTokens extends Procedure<IssueTokensProcedureArgs, Shareholder[]> {
+export const createRefreshSecurityTokenFactoryResolver = (
+  factories: Factories,
+  securityTokenId: string
+) => async () => {
+  return factories.securityTokenFactory.refresh(securityTokenId);
+};
+
+export class IssueTokens extends Procedure<IssueTokensProcedureArgs, Shareholder[] | void> {
```
Comment (monitz87) [inline src/procedures/IssueTokens.ts]: Why did you add `void` to the procedure's return type?

File: src/procedures/CreateCheckpoint.ts (hunk 1 above)
Comment (victorwiebe) [reply src/procedures/CreateCheckpoint.ts]: Due to resolvers. I have removed it I have found the correct way to solve it now

File: src/procedures/IssueTokens.ts (hunk 2 above)
Comment (victorwiebe) [reply src/procedures/IssueTokens.ts]: Due to resolvers, I have found the right way to solve now

File: src/entities/SecurityToken/SecurityToken.ts [hunk 3]
```diff
@@ -26,6 +27,12 @@ export interface Params {
   name: string;
   address: string;
   owner: string;
+  tokenDetails?: string;
```
[the remark this answers is not in the document]
Comment (victorwiebe) [reply src/entities/SecurityToken/SecurityToken.ts]: Fixing..

File: src/utils/index.ts [hunk 4]
```diff
@@ -238,3 +239,23 @@ export const findEvents: FindEvents = ({
...
+      .join('.')
+  ) {
+    case Version.V3_0_0: {
+      return Version.V3_0_0;
+    }
+    case Version.V3_1_0: {
+      return Version.V3_1_0;
+    }
+    default: {
+      throw new Error('Incorrect contract version');
```
Comment (monitz87) [inline src/utils/index.ts]: Should be 
```
new PolymathError({ code: ErrorCode.FatalError, message: `Unsupported Security Token version. Expected 3.0.0 or 3.1.0, got ${version}` });
```
